src/components/dialogs/SpoolmanChangeSpoolDialog.vue

```diff
@@ -0,0 +1,167 @@
+                        style="max-width: 300px" />
+                    <v-spacer />
+                    <v-btn
+                        :title="$t('Panels.SpoolmanPanel.Refresh')"
+                        class="px-2 minwidth-0 ml-3"
+                        :loading="loadings.includes('refreshSpools')"
+                        @click="refreshSpools">
+                        <v-icon>{{ mdiRefresh }}</v-icon>
+                    </v-btn>
+                    <v-btn
+                        :title="$t('Panels.SpoolmanPanel.OpenSpoolManager')"
+                        class="px-2 minwidth-0 ml-3"
+                        @click="openSpoolManager">
+                        <v-icon>{{ mdiDatabase }}</v-icon>
+                    </v-btn>
+                </v-card-title>
+                <v-card-text class="px-0 pb-0">
+                    <v-data-table
+                        :headers="headers"
+                        :items="spools"
+                        item-key="id"
+                        class="elevation-1"
+                        :search="search"
+                        :custom-filter="customFilter"
+                        :disable-sort="true">
+                        <template #no-data>
+                            <div class="text-center">{{ $t('Panels.SpoolmanPanel.NoSpools') }}</div>
+                        </template>
+                        <template #no-results>
+                            <div class="text-center">{{ $t('Panels.SpoolmanPanel.NoResults') }}</div>
+                        </template>
+
+                        <template #item="{ item }">
+                            <SpoolmanChangeSpoolDialogRow :key="item.id" :spool="item" @set-spool="setSpool" />
+                        </template>
+                    </v-data-table>
+                </v-card-text>
+            </panel>
+        </v-dialog>
+    </div>
+</template>
+
+<script lang="ts">
+import Component from 'vue-class-component'
+import { Mixins, Prop } from 'vue-property-decorator'
+import BaseMixin from '@/components/mixins/base'
+import Panel from '@/components/ui/Panel.vue'
+import { mdiCloseThick, mdiAdjust, mdiDatabase, mdiMagnify, mdiRefresh } from '@mdi/js'
+import { ServerSpoolmanStateSpool } from '@/store/server/spoolman/types'
+import SpoolmanChangeSpoolDialogRow from '@/components/dialogs/SpoolmanChangeSpoolDialogRow.vue'
+@Component({
+    components: { SpoolmanChangeSpoolDialogRow, Panel },
+})
+export default class SpoolmanChangeSpoolDialog extends Mixins(BaseMixin) {
+    mdiAdjust = mdiAdjust
+    mdiCloseThick = mdiCloseThick
+    mdiDatabase = mdiDatabase
+    mdiMagnify = mdiMagnify
+    mdiRefresh = mdiRefresh
+
+    @Prop({ required: true }) declare readonly showDialog: boolean
+
+    search = ''
+
+    get spools(): ServerSpoolmanStateSpool[] {
+        const spools = this.$store.state.server.spoolman.spools ?? []
+
+        return spools.filter((spool: ServerSpoolmanStateSpool) => {
+            return spool.filament !== null
```

Comment:
Is there any case where filament can be null? On the spoolman interface the filament field is required to add a spool.

Reply:
ok. look fixed in spoolman. i think this was possible, when you delete only the filament and not the spool. This is now not possible.